src/authentication_manager.rs

```diff
@@ -2,7 +2,7 @@ use async_trait::async_trait;
 use tokio::sync::Mutex;
 
 use crate::custom_service_account::CustomServiceAccount;
-use crate::default_authorized_user::DefaultAuthorizedUser;
+use crate::default_authorized_user::ConfigDefaultCredentials;
```

Comment:
Nit: I would have picked `DefaultApplicationCredentials` to be in line with the wording Google uses. Makes it easier when you read Google docs while using this crate.

Reply:
Good call, changed it.